docs/generate-time-series/javascript.rst

```diff
@@ -0,0 +1,232 @@
+.. _gen-ts-javascript:
+
+==========================================
+Generate time series data using Javascript
+==========================================
+
+This tutorial will show you how to generate some :ref:`experimental time series
+data <gen-ts>` by sampling your `system load`_ using Javascript, specifically `Node.js`_.
+
+.. SEEALSO::
+
+    :ref:`gen-ts`
+
+.. rubric:: Table of contents
+
+.. contents::
+   :local:
+
+
+Prerequisites
+=============
+
+CrateDB must be :ref:`installed and running <install-run>`.
+
+Make sure you're running up-to-date Node/npm.
+
+The `node-postgres`_ driver is our officially supported driver for Javascript.
+With this in mind, use `npm`_ to install the `pg`_ library:
+
+.. code-block:: console
+
+    $ npm install pg
+
+The rest of this tutorial is designed for Node.js's `interactive REPL mode`_ so that
+you can experiment with the commands as you see fit. This tutorial also makes use of
+libraries that operate with `Promises`_, and so starting the interactive node shell
+with support for ``await`` is strongly advised:
+
+.. code-block:: console
+
+    $ node --experimental-repl-await
+
+
+Use Javascript to read your system load
+=======================================
+
+Start an interactive Node session using the ``node`` command. You can read the current system 
+load averages with the `os.loadavg`_ function::
+
+    > const os = require('os')
+    > os.loadavg()
+    [ 0.9794921875, 1.08740234375, 0.97314453125 ]
+
+
+The function returns a triple of `numbers`_ corresponding the average system load over the
+last 1, 5 and 15 minutes.
+
+Set up CrateDB
+==============
+
+First, import the `pg`_ client::
+
+    > const { Client } = require('pg')
+
+Then, `connect`_ to CrateDB, using the port for the `Postgres Wire Protocol`_::
```

Comment:
I suppose not - I just thought it's better to be explicit here than implicit, since the other tutorials use `4200`. I can remove it if you think its necessary?

I'd prefer to keep the port argument in the connection string though, again, for the sake of explicitness.

Reply:
hmm. I see. why do the other tutorials use 4200 (besides the obvious answer that that's how I wrote them lol)? I went with the defaults in the docs for those clients. does that imply those clients not using the postgresql wire protocol?

Reply:
Yep! The python client communicates over 4200, our HTTP port. Our own-written clients don't use the pg wire protocol.